### PR TITLE
+str #17638: Add preStart/postStop to FlexiRoute and FlexiMerge

### PR DESCRIPTION
**Pull request by drewhk**

### CODE REVIEW

**Comment by drewhk** [comment]
PLS BUILD

**Comment by ktoso** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala [hunk 1]
```diff
@@ -31,6 +31,16 @@ private[akka] class FlexiRouteImpl[T, S <: Shape](_settings: ActorFlowMaterializ
   // needed to ensure that at most one element is emitted from onInput
   private val emitted = Array.ofDim[Boolean](outputCount)
 
+  override def preStart(): Unit = {
+    super.preStart()
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala]: was tempted to wrap in try/finally yet currently we know it's the Actor#preStart so not needed I guess

**Comment by drewhk** [comment]
Fixes #17638

File: akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala (hunk 1 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala]: if the super throws then we should not attempt the stage preStart IMHO. (i.e. we did not start the stage)

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/FlexiRouteImpl.scala]: that reasoning also makes sense, good :)

**Pull request merged**